### ISSUE TITLE
bash autocomplete crashes when using sudo

### ISSUE THREAD

**Issue by mtibben** [aws/aws-cli#1521]
When using the aws autocomplete with bash, it crashes when using sudo.

To reproduce:
1. Install aws bash autocomplete
2. Type `sudo aws ` then press tab for autocomplete

Expected result: autocomplete info shown

Actual result:
```
$ sudo aws -bash: compgen: warning: -C option may not work as you expect
Traceback (most recent call last):
  File "/usr/local/Cellar/awscli/1.8.6/libexec/bin/aws_completer", line 25, in <module>
    awscli.completer.complete(cline, cpoint)
  File "/usr/local/Cellar/awscli/1.8.6/libexec/lib/python2.7/site-packages/awscli/completer.py", line 166, in complete
    choices = Completer().complete(cmdline, point)
  File "/usr/local/Cellar/awscli/1.8.6/libexec/lib/python2.7/site-packages/awscli/completer.py", line 155, in complete
    self._process_command_line()
  File "/usr/local/Cellar/awscli/1.8.6/libexec/lib/python2.7/site-packages/awscli/completer.py", line 126, in _process_command_line
    self.current_word = self.words[-1]
IndexError: list index out of range
```

Version info:
```
$ aws --version
aws-cli/1.8.6 Python/2.7.10 Darwin/14.5.0
$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)
Copyright (C) 2007 Free Software Foundation, Inc.
```

**Comment by kyleknap** [comment]
I am having troubles replicating this. Can you describe how you install the CLI and the tab completer? Ideally a reproducible set of steps that I could follow.

**Comment by JordonPhillips** [comment]
This issue is being closed due to inactivity

**Comment by mtibben** [comment]
Hi @kyleknap @JordonPhillips, Here are more detailed instructions to help you reproduce the problem on Mac OSX:

1. Install [homebrew](http://brew.sh/)
2. Install awscli
```
brew install bash-completion
brew install awscli
```

Add the following lines to your ~/.bash_profile:
```
  if [ -f $(brew --prefix)/etc/bash_completion ]; then
    . $(brew --prefix)/etc/bash_completion
  fi
```
Add the following lines to your ~/.bashrc
```
  complete -C aws_completer aws
```
3. Launch a bash shell
```
bash
```

4. Type `sudo aws` then press tab for autocomplete

**Comment by mtdowling** [comment]
I was able to reproduce this issue when I placed the `complete -C aws_completer aws` command in my `~/.bash_rc` using `sudo aws`. I also receive this cryptic error message (the same you received):

```
-bash: compgen: warning: -C option may not work as you expect
```

This error is somehow caused by the `bash-completion` package not playing well with the completion defined in `~/.bashrc` rather than your `~/.bash_profile`. This has something to do with how complete and compgen share some common code paths.

Moving the completion to `~/.bash_profile` removed the crash, but did not use the custom CLI autocompleter (it just listed files in the working directory). The same thing happened (no crash, but no custom completion), when I had it installed but defined my autocompletion in `~/.bash_profile`.

In order to make this work, I just added an autocomplete rule for `aws` and `sudo aws` in my `~/.bash_profile`:

```
complete -C "$(brew --prefix)/bin/aws_completer" aws
complete -C "$(brew --prefix)/bin/aws_completer" sudo aws
```
